esileapclient/common/utils.py

```diff
@@ -0,0 +1,56 @@
+import re
+import operator
+
+OPS = {
+    '>=': operator.ge,
+    '<=': operator.le,
+    '>': operator.gt,
+    '<': operator.lt,
+    '=': operator.eq,
+}
+
+
+def parse_property_filter(filter_str):
+    """Parse a property filter string into a key, operator, and value."""
+    match = re.match(r'([^><=]+)([><=]+)(.+)', filter_str)
+    if not match:
+        raise ValueError(f"Invalid property filter format: {filter_str}")
+    key, op_str, value_str = match.groups()
+    return key, OPS[op_str], value_str
+
+
+def convert_value(value_str):
+    """Convert a value string to an appropriate type for comparison."""
+    try:
+        return int(value_str)
+    except ValueError:
+        try:
+            return float(value_str)
+        except ValueError:
+            return value_str
+
+
+def filter_nodes_by_properties(nodes, properties):
+    """Filter a list of nodes based on property filters."""
+    if not properties:
+        return nodes
+    property_filters = []
+    for prop in properties:
+        key, op, value_str = parse_property_filter(prop)
+        value = convert_value(value_str)
```

Comment:
would it make sense to run `convert_value` within `parse_property_filter`? then you can do something like `property_filters = list(map(parse_property_filter, properties))`

Reply:
Yes, I will do that for better code readability.